From 3cda04ffb40ec461c1797d74eaad1703922a8b2c Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Wed, 23 Jan 2013 09:44:20 -0800
Subject: [PATCH 333/337] MIPS/OCTEON, netdev/octeon-ethernet: Fix error bit
 notifier priorities.

Based On SDK 3.0.0-482

The octeon-error-tree notifier should be the lowest priority, not the
highest.

Signed-off-by: David Daney <david.daney@cavium.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
Signed-off-by: Jack Tan <jack.tan@windriver.com>
---
 arch/mips/cavium-octeon/octeon-error-tree.c  |    2 +-
 drivers/net/ethernet/octeon/ethernet-rgmii.c |    2 +-
 drivers/net/ethernet/octeon/ethernet-spi.c   |    2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/mips/cavium-octeon/octeon-error-tree.c b/arch/mips/cavium-octeon/octeon-error-tree.c
index 9b1aa9f..8009022 100644
--- a/arch/mips/cavium-octeon/octeon-error-tree.c
+++ b/arch/mips/cavium-octeon/octeon-error-tree.c
@@ -267,7 +267,7 @@ static int __init octeon_error_tree_init(void)
 	octeon_error_tree_current = tree->tree;
 
 	octeon_error_tree_notifier.notifier_call = octeon_error_tree_hw_status;
-	octeon_error_tree_notifier.priority = 10;
+	octeon_error_tree_notifier.priority = 0;
 	octeon_hw_status_notifier_register(&octeon_error_tree_notifier);
 
 	return octeon_error_tree_enable(CVMX_ERROR_GROUP_INTERNAL, -1);
diff --git a/drivers/net/ethernet/octeon/ethernet-rgmii.c b/drivers/net/ethernet/octeon/ethernet-rgmii.c
index 9b26789..92a7407 100644
--- a/drivers/net/ethernet/octeon/ethernet-rgmii.c
+++ b/drivers/net/ethernet/octeon/ethernet-rgmii.c
@@ -206,7 +206,7 @@ int cvm_oct_rgmii_open(struct net_device *dev)
 	 * problems if the PHY doesn't implement inband status
 	 * properly.
 	 */
-	priv->hw_status_notifier.priority = 1;
+	priv->hw_status_notifier.priority = 10;
 	priv->hw_status_notifier.notifier_call = cvm_oct_rgmii_hw_status;
 	octeon_hw_status_notifier_register(&priv->hw_status_notifier);
 
diff --git a/drivers/net/ethernet/octeon/ethernet-spi.c b/drivers/net/ethernet/octeon/ethernet-spi.c
index f603f7f..64e30b2 100644
--- a/drivers/net/ethernet/octeon/ethernet-spi.c
+++ b/drivers/net/ethernet/octeon/ethernet-spi.c
@@ -151,7 +151,7 @@ int cvm_oct_spi_init(struct net_device *dev)
 
 	if ((priv->ipd_port == 0) || (priv->ipd_port == 16)) {
 
-		priv->hw_status_notifier.priority = 1;
+		priv->hw_status_notifier.priority = 10;
 		priv->hw_status_notifier.notifier_call = cvm_oct_spi_hw_status;
 		r = octeon_hw_status_notifier_register(&priv->hw_status_notifier);
 		if (r)
-- 
1.7.5.4

